File: vendror_android/app/src/main/java/magentoegypt/locafy/base_app/base/BaseViewModel.java
package magentoegypt.locafy.base_app.base;

import androidx.annotation.NonNull;
import androidx.lifecycle.LiveData;
import androidx.lifecycle.ViewModel;

public class BaseViewModel<E extends RequestWatcher> extends ViewModel {
    E repo;
    public void setRepo(@NonNull E repo) {
        this.repo = repo;
        loading = repo._loading;
        error = repo._error;
    }
    public LiveData<Boolean> loading;
    public LiveData<String> error;
    public LiveData<Boolean> getLoading() {
        return loading;
    }

    public LiveData<String> getError() {
        return error;
    }
}

File: vendror_android/app/src/main/java/magentoegypt/locafy/addons/advance_Report/appBase/BaseViewModel.kt
package magentoegypt.locafy.addons.advance_Report.appBase

import androidx.lifecycle.LiveData
import androidx.lifecycle.ViewModel


abstract class BaseViewModel<T : BaseAppRepo?> : ViewModel() {
    private var repo: T? = null
    var error: LiveData<String>? = null
    var loading: LiveData<Boolean>? = null
    fun setRepo(repo: T) {
        this.repo = repo
        error = repo!!._error
        loading = repo._loading
    }
}
